### PR TITLE
Adding types for whois-json

### PR DESCRIPTION
**Pull request by hwaien**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/whois-json/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+     * The SOCKS4 protocol.
+     */
+    SOCKS4 = 4,
+    /**
+     * The SOCKS5 protocol.
+     *
+     * @see - https://tools.ietf.org/html/rfc1928
+     */
+    SOCKS5 = 5
+}
```
Comment (DanielRosenwasser) [inline types/whois-json/index.d.ts:75]: This declaration doesn't exist at runtime.

File: types/whois-json/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+// Definitions by: Hwai-En Ho <https://github.com/hwaien>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Perform a WHOIS lookup.
+ *
+ * @param domainName - The domain name on which to perform the lookup.
+ * @param options - Options to tweak the behavior of a WHOIS lookup.
+ */
+export default function whois(domainName: string, options?: WhoisOptions): Promise<WhoisResult>;
```
Comment (DanielRosenwasser) [inline types/whois-json/index.d.ts:12]: This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Variables

```ts
declare var xyz: SomeType;

export = xyz;
```

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

Comment (hwaien) [reply types/whois-json/index.d.ts:12]: Thanks for the pointer!

I rewrote the file using the following as example: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/a2hs.js/index.d.ts

File: types/whois-json/index.d.ts (hunk 1 above)
Comment (hwaien) [reply types/whois-json/index.d.ts]: You are right that it doesn't exist at runtime. I was thinking it is a useful aid at development time to let developers know that only the values 4 and 5 are valid arguments.

Comment (DanielRosenwasser) [reply types/whois-json/index.d.ts]: Yeah - I agree that that would've been nice, but unfortunately this is going to result in users trying to reference a value (`SocksProtocolVersion`) that doesn't actually exist. It also won't allow users to write `4` or `5` directly.

I'd say the unfortunate best that we can ship is something like this:

```ts
type SocksProtocolVersion = 4 | 5;
```

Comment (hwaien) [reply types/whois-json/index.d.ts]: Oh didn't know this syntax. I will update it accordingly. On a related note: is there a way to learn TypeScript syntax other than from the handbook? I tried reading the language spec but saw that it is several years old. Any advice is appreciated @DanielRosenwasser :)

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by hwaien** [comment]
Ready to merge

**Pull request merged**